### PR TITLE
dbcp updated

### PR DESCRIPTION
**Pull request by anandh619**

### CODE REVIEW

**Comment by jchristopherinc** [comment]
add `.gitignore` to the project and ignore files from `.idea`

File: pingdom/src/main/java/com/pulse/dbcp/DataBaseUtility.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package com.pulse.dbcp;
+
+import org.apache.commons.dbcp2.BasicDataSource;
+
+public class DataBaseUtility {
+
+
+    private static final String DRIVER_CLASS_NAME = "org.postgresql.Driver";
```
Comment (jchristopherinc) [inline pingdom/src/main/java/com/pulse/dbcp/DataBaseUtility.java]: @nandini2052 experimented something to move the DB URL, name, password to properties. Why don't you check it out for this also?

File: pingdom/src/main/java/com/pulse/dbcp/DataBaseInit.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        ResultSet resultSet = null;
+        try {
+
+            BasicDataSource bds = DataBaseUtility.getInstance().getBds();
+            connection = bds.getConnection();
+            //looping through array of objects of type URLPayload and inserting into database
+            for(URLPayload uObj:urlObj) {
+
+
+                String sql = "insert into URLS(user_id,url,created_at,status) values(?,?,?,?)";
```
Comment (jchristopherinc) [inline pingdom/src/main/java/com/pulse/dbcp/DataBaseInit.java]: Is this some sort of dummy query to check if DB connection is working?

File: pingdom/src/main/java/com/pulse/dbcp/DataBaseUtility.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    private static final String DB_URL = "jdbc:postgresql://localhost/pulse";
+    //enter your local database username
+    private static final String DB_USER = "db_user";
+    //password for your respective db
+    private static final String DB_PASSWORD = "db_user_password";
+
+    private static final int CONN_POOL_SIZE = 10;
+
+
+    private BasicDataSource bds = new BasicDataSource();
```
Comment (jchristopherinc) [inline pingdom/src/main/java/com/pulse/dbcp/DataBaseUtility.java]: `Bds`?  u can name this object as `basicDataSource`, there by we can have more context on the object when we read the code

**Review by jchristopherinc — CHANGES_REQUESTED** [review]
Requesting changes..

add `.gitignore` to the project and ignore files from `.idea`

File: pingdom/src/main/java/com/pulse/dbcp/DataBaseInit.java (hunk 2 above)
Comment (anandh619) [reply pingdom/src/main/java/com/pulse/dbcp/DataBaseInit.java]: No. I'm inserting the urls into DB. User_id is dummy here. I'll update it after getting it from cookie.

File: pingdom/src/main/java/com/pulse/dbcp/DataBaseUtility.java (hunk 3 above)
Comment (anandh619) [reply pingdom/src/main/java/com/pulse/dbcp/DataBaseUtility.java]: ok I'll change it.

File: pingdom/src/main/java/com/pulse/controller/UrlController.java [hunk 4]
```diff
@@ -17,14 +18,15 @@
 
 @Path("/url")
 public class UrlController {
+    private final static Logger log = LoggerFactory.getLogger("global");
```
Comment (jchristopherinc) [inline pingdom/src/main/java/com/pulse/controller/UrlController.java]: change it to class name

File: pingdom/src/main/java/com/pulse/dbcp/UrlDBInsert.java [hunk 5]
```diff
@@ -6,52 +6,45 @@
...
-    public DataBaseInit(ArrayList<URLPayload> urlList) {
+    public UrlDBInsert(ArrayList<URLPayload> urlList) {
         this.urlObj = urlList;
-        System.out.println("constructor has been called");
     }
 
 
-    public void dbConnect(){
-        System.out.println("Enters the dbConnect");
+    public void insertQuery(){
```
Comment (jchristopherinc) [inline pingdom/src/main/java/com/pulse/dbcp/UrlDBInsert.java]: name your methods based on it's functionality. Here it inserts URLs into the table. So, you can name it `insertUrlsList`

File: pingdom/src/main/java/com/pulse/dbcp/UrlDBInsert.java [hunk 6]
```diff
@@ -6,52 +6,45 @@
...
-
-            BasicDataSource bds = DataBaseUtility.getInstance().getBds();
-            connection = bds.getConnection();
-//            statement = connection
-//                    .createStatement();
-//            statement.executeUpdate("INSERT  into URLS(user_id,url,created_at,status) VALUES (1,'http://www.facebook.com','1970-01-01 00:00:01','up')");
-
+            log.info("Entered into the try block");
+            BasicDataSource basicDataSource = DBInit.getInstance().getBds();
+            connection = basicDataSource.getConnection();
```
Comment (jchristopherinc) [inline pingdom/src/main/java/com/pulse/dbcp/UrlDBInsert.java]: Have a helper class to just give you the connection.

File: pingdom/src/main/java/com/pulse/service/EmailService.java [hunk 7]
```diff
@@ -10,6 +11,8 @@
 import javax.mail.internet.MimeMessage;
 
 public class EmailService {
+    private final static Logger log = LoggerFactory.getLogger("global");
```
Comment (jchristopherinc) [inline pingdom/src/main/java/com/pulse/service/EmailService.java]: global what :P

**Review by jchristopherinc — COMMENTED** [review]
Minor comments. can merge after fixing.

**Comment by jchristopherinc** [comment]
Merged as part of #6

**Pull request closed without merging**